File: orchalanguage-spring-integration-implementation/src/main/java/orcha/lang/compiler/referenceimpl/springIntegration/WhenInstructionFactory.java
package orcha.lang.compiler.referenceimpl.springIntegration;

import org.springframework.beans.factory.config.AbstractFactoryBean;

public class WhenInstructionFactory extends AbstractFactoryBean<WhenInstructionForSpringIntegration> {

    public WhenInstructionFactory(){
        setSingleton(false);
    }

    @Override
    public Class<?> getObjectType() {
        return WhenInstructionForSpringIntegration.class;
    }

    @Override
    protected WhenInstructionForSpringIntegration createInstance() throws Exception {
        return new WhenInstructionForSpringIntegration();
    }
}

File: orchalang-kafka-implementation/src/main/kotlin/orcha/lang/compiler/kafka/WhenInstructionFactory.kt
package orcha.lang.compiler.kafka

import org.springframework.beans.factory.config.AbstractFactoryBean

class WhenInstructionFactory : AbstractFactoryBean<WhenInstructionForSpringIntegration>() {
    init {
        isSingleton = false
    }

    override fun getObjectType(): Class<*>? {
        return WhenInstructionForSpringIntegration::class.java
    }

    @Throws(Exception::class)
    override fun createInstance(): WhenInstructionForSpringIntegration {
        return WhenInstructionForSpringIntegration()
    }
}
